packages/base/lua/Utils.lua

```diff
@@ -576,6 +576,49 @@ function forEachDirectory(directory, _sof_)
 	end)
 end
 
+local function getFilesRecursively(directory)
+	local files = {}
+
+	if not directory:exists() then return files end
+
+	_Gtme.forEachDirectory(directory, function(dir)
+		for _, v in ipairs(getFilesRecursively(dir)) do
+			table.insert(files, v)
+		end			
+	end)
+ 
+	_Gtme.forEachFile(directory, function(file)
+		table.insert(files, file)
+	end)
+	
+	return files
+end
+
+--- Second order function to transverse a given directory recursively,
+-- applying a given function on each of its internal files.
+-- If any of the function calls returns
+-- false, forEachRecursiveDirectory() stops and returns false, otherwise it returns true.
+-- @arg directory A string with the path to a directory, or a base::Directory.
+-- @arg _sof_ A user-defined function that takes a file path as argument.
+-- @usage forEachRecursiveDirectory(packageInfo("base").path.."data", function(file)
+--     print(file)
+-- end)
+function forEachRecursiveDirectory(directory, _sof_)
+	local dir = directory
+
+	if type(dir) ~= "Directory" then
+		if type(dir) == "string" then
+			dir = Directory(dir)
+		else
+			customError("The directory argument must be a 'Directory' or 'string' path.")
```

Comment:
It should be 

`Argument '#1' must be ...`

The user does not know the name of the argument.